setup/Core/SetupOperation.cs

```diff
@@ -151,44 +156,33 @@ private static bool DeleteEmptyDirsRecursion(string dir) {
 			return true;
 		}
 
-		protected readonly ITaskInterface taskInterface;
-		protected int progress;
-
-		protected SetupOperation(ITaskInterface taskInterface) {
-			this.taskInterface = taskInterface;
-		}
-
 		/// <summary>
-		/// Run the task, any exceptions thrown will be written to a log file and update the status label with the exception message
+		///     Run the task, any exceptions thrown will be written to a log file and update the status label with the exception
+		///     message
 		/// </summary>
-		public abstract void Run();
+		public abstract Task Run(IProgress progress, CancellationToken cancellationToken = default);
 
 		/// <summary>
-		/// Display a configuration dialog. Return false if the operation should be cancelled.
+		///     Display a configuration dialog. Return false if the operation should be cancelled.
 		/// </summary>
-		public virtual bool ConfigurationDialog() => true;
+		/// <param name="cancellationToken"></param>
+		public virtual ValueTask<bool> ConfigurationPrompt(CancellationToken cancellationToken = default) => ValueTask.FromResult(true);
 
 		/// <summary>
-		/// Display a startup warning dialog
+		///     Display a startup warning dialog
 		/// </summary>
 		/// <returns>true if the task should continue</returns>
 		public virtual bool StartupWarning() => true;
 
 		/// <summary>
-		/// Will prevent successive tasks from executing and cause FinishedDialog to be called
+		///     Will prevent successive tasks from executing and cause FinishedDialog to be called
 		/// </summary>
 		/// <returns></returns>
 		public virtual bool Failed() => false;
 
 		/// <summary>
-		/// Will cause FinishedDialog to be called if warnings are not supressed
-		/// </summary>
-		/// <returns></returns>
-		public virtual bool Warnings() => false;
```

Comment:
auto setup (for CI) isn't allowed to complete with warnings. Please account for this.

Reply:
Perhaps a --strict CLI arg

Reply:
The CLI exits with status code 1 now if the task failed or the strict option is passed to `TaskRunner` and the task has warnings. I did not add an additional CLI arg for this, but just passed `strict: true` to `TaskRunner` in the `SetupAutoCommand`.